lib/msf/core/post/windows/services.rb

```diff
@@ -197,32 +226,60 @@ def each_service(&block)
   # @todo Rewrite to allow operating on a remote host
   #
   def service_list
-    if load_extapi
-      return session.extapi.service.enumerate
-    else
-      serviceskey = "HKLM\\SYSTEM\\CurrentControlSet\\Services"
-      a =[]
-      services = []
-      keys = registry_enumkeys(serviceskey)
-      keys.each do |s|
-        if a.length >= 10
-          a.first.join
-          a.delete_if {|x| not x.alive?}
-        end
-        t = framework.threads.spawn(self.refname+"-ServiceRegistryList",false,s) { |sk|
-          begin
-            srvtype = registry_getvaldata("#{serviceskey}\\#{sk}","Type").to_s
-            if srvtype == "32" or srvtype == "16"
-              services << {:name => sk }
-            end
-          rescue
+    return meterpreter_service_list if session.type == 'meterpreter'
+
+    services = []
+    each_service do |s|
+      services << {:name => s}
+    end
+
+    services
+  end
+
+  def meterpreter_service_list
```

Comment:
Missing RDoc documentation for this new function.

Reply:
Marking this as resolved in favor of https://github.com/rapid7/metasploit-framework/pull/16928#discussion_r966440527